testsuite/features/support/env.rb

```diff
@@ -150,7 +150,12 @@ def capybara_register_driver
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
   if scenario.failed?
     begin
-      if web_session_is_active?
+      if scenario.exception.is_a?(Selenium::WebDriver::Error::WebDriverError)
+        log "Caught web driver error: #{scenario.exception.message}"
+        Capybara.current_session.driver.quit
```

Comment:
What happen when Capybara.current_session is broken?

Reply:
This is explained in PR's description.

As a side note, it's not only the current session that is broken, but the whole webdriver.

Reply:
What I mean is that if `Capybara.current_session` is broken, you will not be able to call `quit`.